lib/util/search/address_search_field.dart

```diff
@@ -7,18 +7,18 @@ import 'address_suggestion.dart';
 class AddressSearchField extends StatelessWidget {
   final AddressType addressType;
   final TextEditingController controller;
-  final void Function(AddressSuggestion) onSelected;
+  final void Function(AddressSuggestion)? onSelected;
```

Comment:
I don't understand this change? seems unneeded

Reply:
Seeing this as a general class that could be used outside of our context with suggestion objects that have to be set, I see no reason why you would have to provide a function here. In most cases, a controller would suffice

Reply:
I'm sorry, but I don't agree - the whole thing does not make sense if we don't have `AddressSuggestion` objects - it's literally what the `SearchField` was made for

Reply:
Is there a case where we would need it without an object?

Reply:
Why not give users the choice? If I just want a "Choose your favorite place in the world" textfield and just wanted to store the string, I feel like I should be able to do that without needing to provide a (){} function

Reply:
We are not writing a library for other people, we are writing an app the we use ourselves...